amethyst_renderer/src/mouse.rs

```diff
@@ -1,34 +1,42 @@
 //! Util functions that change how the mouse moves and looks.
 
 use super::WindowMessages;
-use winit::{CursorState, MouseCursor};
+use winit::MouseCursor;
 
 /// Hide the cursor, so it's invisible while playing. Can't be used at the same time as grab_cursor.
 pub fn hide_cursor(msg: &mut WindowMessages) {
-    change_cursor_state(msg, CursorState::Hide);
+    msg.send_command(move |win| win.hide_cursor(true));
 }
 
 /// Set the cursor back to normal/visible.
 pub fn release_cursor(msg: &mut WindowMessages) {
-    change_cursor_state(msg, CursorState::Normal);
+    msg.send_command(move |win| {
+        if let Err(err) = win.grab_cursor(false) {
+            error!("Unable to release the cursor. Error: {:?}", err);
+        }
+        win.hide_cursor(false);
+    });
 }
 
 /// Grab the cursor to prevent it from going outside the screen.
 pub fn grab_cursor(msg: &mut WindowMessages) {
-    change_cursor_state(msg, CursorState::Grab);
-}
-
-fn change_cursor_state(msg: &mut WindowMessages, state: CursorState) {
     msg.send_command(move |win| {
-        if let Err(err) = win.set_cursor_state(state) {
-            error!("Unable to change the cursor state! Error: {:?}", err);
+        if let Err(err) = win.grab_cursor(true) {
+            error!("Unable to grab the cursor. Error: {:?}", err);
         }
     });
 }
 
-/// Hide the cursor, so it's invisible while player. Can be used at the same time as grab_cursor.
+/// Hide the cursor, so it's invisible.
+///
+/// This grabs the cursor and hides it, so callers do not have to make separate calls.
 pub fn set_mouse_cursor_none(msg: &mut WindowMessages) {
-    set_mouse_cursor(msg, MouseCursor::NoneCursor);
+    msg.send_command(move |win| {
+        if let Err(err) = win.grab_cursor(true) {
+            error!("Unable to grab the cursor. Error: {:?}", err);
+        }
+        win.hide_cursor(true);
+    });
```

Comment:
Hiding and grabbing are not the same thing. They should stay separated

Reply:
#rookie_move
fixing